### DISCUSSION TITLE
Doubts about compare_models and the scores

### DISCUSSION THREAD

**Discussion by etiur** [pycaret/pycaret#3737] [Q&A] [answered]
Hello, I am new to PyCaret so I would like know more about it, could you answer a few questions for me thank you very much:

1. Are the scores returned from the compare_models a mean of the training CV score and the Validation CV score from the different folds?
2. Is there something similar to return_train_score argument in compare_models so I can know which model performed better at the validation?
3. Is it possible to have even more detailed information like the scores per class for example so I can see the precision on class 1 instead of the overall precision? I know that I can get the plot for that.
4. Does the compare_model tune the hyperparameters? 
5. If it doesn't tune the models' parameters does it mean that I could get better results by calling tune_model on the best model returned from compare_models?
6. Are the models from compare_models well calibrated or do I need to call calibrate_model.

**Comment by AlexShyXie** [comment]
Me too, I want to look the performance in training data to find it is overfitting or not

**Comment by moezali1** [comment, marked as answer]
- Are the scores returned from the compare_models a mean of the training CV score and the Validation CV score from the different folds? **Yes it is.**
- Is there something similar to return_train_score argument in compare_models so I can know which model performed better at the validation? **No, there isn't.**
- Is it possible to have even more detailed information like the scores per class for example so I can see the precision on class 1 instead of the overall precision? I know that I can get the plot for that. **No, unless you create and add custom metric to do that.**
- Does the compare_model tune the hyperparameters? **No, tune_model does hyperparameter tuning.**
- If it doesn't tune the models' parameters does it mean that I could get better results by calling tune_model on the best model returned from compare_models? **Yes, you can.**
- Are the models from compare_models well calibrated or do I need to call calibrate_model. **There are not calibrated. You can use calibrate_model function if you want to calibrate.**
